test/tools/router_check/router.cc

```diff
@@ -0,0 +1,204 @@
+#include "test/tools/router_check/router.h"
+
+Json::ObjectPtr RouterCheckTool::loadJson(const std::string& config_json,
+                                          const std::string& schema) {
+
+  try {
+    // Json configuration
+    Json::ObjectPtr loader = Json::Factory::LoadFromFile(config_json);
+    loader->validateSchema(schema);
+    return loader;
+  } catch (const EnvoyException& ex) {
```

Comment:
You have quite a bit of extra error handling after you catch this exception. Embrace exceptions, they are awesome. In this case, I would have a top level try/catch in main(), and catch EnvoyException, print an error, and exit(1). If you do this, you can get rid of quite a bit of extraneous error handling in the code.